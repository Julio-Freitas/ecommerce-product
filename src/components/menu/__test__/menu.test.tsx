import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import useEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import Menu from "..";

const renderWithRouter = (ui: any, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);
  return {
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

describe("Menu", () => {
  it("Should be Menu", () => {
    const { container } = renderWithRouter(<Menu />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should active item menu when change pathname", async () => {
    renderWithRouter(<Menu />)
    const menu = screen.getAllByRole("link");
    const collectionsItem = menu[1];

    await useEvent.click(collectionsItem);
    expect(window.location.pathname).toStrictEqual("/collections");
    expect(collectionsItem).toHaveClass('_container-active_9678f5');
  });
});
