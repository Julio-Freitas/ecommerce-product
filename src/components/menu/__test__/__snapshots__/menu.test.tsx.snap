// Vitest Snapshot v1

exports[`Menu > Should be Menu 1`] = `
<menu
  class="_container_9678f5"
>
  <a
    href="/"
  >
    <img
      alt="blacklogo "
      src="/src/assets/images/logo.svg"
      title="logo"
    />
  </a>
  <nav>
    <a
      class="__9678f5"
      href="/collections"
    >
      Collections
    </a>
    <a
      class="__9678f5"
      href="/men"
    >
      Men
    </a>
    <a
      class="__9678f5"
      href="/woman"
    >
      Woman
    </a>
    <a
      class="__9678f5"
      href="/about"
    >
      About
    </a>
    <a
      class="__9678f5"
      href="/contact"
    >
      Contact
    </a>
  </nav>
  <div
    class="_cart_9bc7a5"
  >
    <button
      class="_cart-icon_9bc7a5"
      data-amount="0"
      type="button"
    >
      <img
        alt="icon cart"
        src="/src/assets/images/icon-cart.svg"
        title="card"
      />
    </button>
    <img
      alt="icon avatar"
      class="_cart-avatar_9bc7a5"
      src="/src/assets/images/image-avatar.png"
      title="Avatar"
    />
    <div
      class="_cart__hidden--info_9bc7a5"
    >
      <h3>
        Card
      </h3>
      <p
        class="_cart__info-empty_9bc7a5"
      >
        Your cart is empty.
      </p>
    </div>
  </div>
</menu>
`;
